flocker/control/_config.py

```diff
@@ -1014,9 +1022,15 @@ def _parse_link_configuration(self, application_name, config):
                     raise ValueError(
                         "Unrecognised keys: {keys}.".format(
                             keys=', '.join(sorted(link))))
-                links.append(Link(local_port=local_port,
-                                  remote_port=remote_port,
-                                  alias=alias))
+                try:
+                    links.append(Link(local_port=local_port,
+                                      remote_port=remote_port,
+                                      alias=alias))
+                except InvariantException:
+                    # We've already verified types of Link parameters, so
+                    # that just leaves the alias naming limits as the
+                    # source of error.
```

Comment:
The port values could be out of range.  Not something we enforce in the `Link` model now but it could be later.  Can we get some useful information out of the raised `InvariantException`?